code/modules/atmospherics/machinery/datum_pipeline.dm

```diff
@@ -355,3 +353,24 @@
 	if(gasmix_color != current_color)
 		gasmix_color = current_color
 		UpdateGasVisuals()
+
+/obj/effect/abstract/gas_visual
+	appearance_flags  = RESET_COLOR | KEEP_APART
+	vis_flags = VIS_INHERIT_ICON_STATE | VIS_INHERIT_LAYER | VIS_INHERIT_PLANE | VIS_INHERIT_ID
+	var/current_color
+	var/color_filter
+
+/obj/effect/abstract/gas_visual/Initialize(mapload)
+	. = ..()
+	color_filter = filter(type="color", color=matrix())
+	filters += color_filter
+	color_filter = filters[filters.len]
+	if(current_color)
+		animate(color_filter, color=current_color, time=5)
+
+/obj/effect/abstract/gas_visual/proc/ChangeColor(new_color)
+	current_color = new_color
+	if(isnull(color_filter))
+		// Called before init
+		return
+	animate(color_filter, time=5, color=new_color)
```

Comment:
tested? color filters animate weird sometimes

Reply:
Yeah I tested and it seemed to be fine just giving it a color instead of a matrix